CONTRIBUTING.md

```diff
@@ -1,161 +1,176 @@
+when proposing changes to the contributor/ directory. <p>
 
-1. Sign up for GitHub (a free account is fine): https://github.com/join
-2. Add your ssh public key to your account: https://github.com/settings/ssh
-3. Hit the "Fork" button on the web page for the Juju repo:
-    https://github.com/juju/juju
+> Tip: If you start from an existing doc on
+https://canonical-juju.readthedocs-hosted.com/en/latest/, you can find its
+source file by scrolling to the bottom of the page and clicking on **Edit this
+page on GitHub** -- though we still recommend you do not make changes there
+directly there but rather in a local clone, as that will make it easier for you
+to iterate during the PR review process.
 
-At this point you will have your own copy under your github account. Note
-that your fork is not automatically kept in sync with the official Juju repo
-(see [Staying in sync](#staying-in-sync)).
 
-Note that Juju has dependencies hosted elsewhere with other version control
-tools.
+If you are adding a new page, make sure to file it correctly and also add it to
+the correct `index.md` page, or else it won't be visible in the published docs.
 
-Staying in sync
----------------
+All changes should follow the existing patterns, including
+  [Diátaxis](https://diataxis.fr), the [Canonical Documentation Style
+  Guide](https://docs.ubuntu.com/styleguide/en), the modular structure, the
+  cross-referencing pattern, [MyST
+  Markdown](https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide-myst/),
+  etc.
 
-Make sure your local copy and GitHub fork stay in sync with upstream:
+Changes should be inspected by building the docs and fixing any issues
+discovered that way. To build, navigate to `juju/docs`, then run `make run` and
+follow the link to preview the docs in a browser. For other checks, see  `make
+[Tab]` and, if you get errors, try `make clean`.
+</details>
 
-```bash
-cd juju
-git pull upstream
-```
+----------------
+
+----------------
+<details>
+
+<summary>Further info: Code</summary>
+
+### Installing Go
 
-Dependency management
-=====================
+`juju` is written in [Go](https://go.dev/). To install Go see [Go
+docs](https://golang.org/doc/install#install).
 
-go mod
-------
+### Building Juju and its dependencies
 
-Juju uses Go modules to manage dependencies.
+Fork and clone the Juju repo, then navigate to the root directory and run `make
+install`:
 
-Updating dependencies
----------------------
+```
+git clone https://github.com/<user>/juju.git
+cd juju
+make install
+```
 
-To update a dependency, use the following, ensuring that the dependency is
-using a version where possible, or a commit hash if not available:
+### Updating Go dependencies
+
+Juju uses Go modules to manage dependencies. To update a dependency, use the
+following, ensuring that the dependency is using a version where possible, or a
+commit hash if not available:
```

Comment:
These lines are a bit confusing

Reply:
Any suggestion? (This is content I borrowed from the previous iteration wholesale.)